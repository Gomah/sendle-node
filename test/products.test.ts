import { describe, expect, it } from 'vitest';
import { client } from './helpers';

describe('Products', () => {
  it('should get a list of products', async () => {
    const products = await client.products.get({
      receiver_country: 'AU',
      receiver_postcode: '2000',
      receiver_suburb: 'SYDNEY',
      sender_country: 'AU',
      sender_postcode: '2000',
      sender_suburb: 'SYDNEY',
      weight_units: 'kg',
      weight_value: '1',
    });

    expect(products[0]).toMatchSnapshot({
      plan: expect.any(String),
      eta: {
        date_range: expect.any(Array),
        days_range: expect.any(Array),
        for_send_date: expect.any(String),
      },
      price_breakdown: {
        base: {
          amount: expect.any(Number),
        },
        base_tax: {
          amount: expect.any(Number),
        },
        fuel_surcharge: {
          amount: expect.any(Number),
        },
        fuel_surcharge_tax: {
          amount: expect.any(Number),
        },
      },
      tax_breakdown: {
        gst: {
          amount: expect.any(Number),
        },
      },
      quote: {
        gross: {
          amount: expect.any(Number),
        },
        net: {
          amount: expect.any(Number),
        },
        tax: {
          amount: expect.any(Number),
        },
      },
    });
  });
});
