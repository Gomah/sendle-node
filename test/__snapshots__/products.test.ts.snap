// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Products > should get a list of products 1`] = `
{
  "allowed_packaging": "any",
  "eta": {
    "date_range": Any<Array>,
    "days_range": Any<Array>,
    "for_send_date": Any<String>,
  },
  "plan": Any<String>,
  "price_breakdown": {
    "base": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "base_tax": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "cover": {
      "amount": 0,
      "currency": "AUD",
    },
    "cover_tax": {
      "amount": 0,
      "currency": "AUD",
    },
    "discount": {
      "amount": 0,
      "currency": "AUD",
    },
    "discount_tax": {
      "amount": 0,
      "currency": "AUD",
    },
    "fuel_surcharge": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "fuel_surcharge_tax": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
  },
  "product": {
    "code": "STANDARD-PICKUP",
    "first_mile_option": "pickup",
    "name": "Standard Pickup",
    "service": "standard",
  },
  "quote": {
    "gross": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "net": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "tax": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
  },
  "route": {
    "description": "Sydney, Australia to Sydney, Australia",
    "type": "same-city",
  },
  "tax_breakdown": {
    "gst": {
      "amount": Any<Number>,
      "currency": "AUD",
      "rate": 0.1,
    },
  },
}
`;
